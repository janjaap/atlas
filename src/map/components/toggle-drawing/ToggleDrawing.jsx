import React from 'react';
import PropTypes from 'prop-types';

import drawToolConfig from '../../services/draw-tool/draw-tool-config';

import IconMeasure from '../../../../public/images/icon-measure.svg';

import './_toggle-drawing.scss';

const ToggleDrawing = ({ drawingMode, shapeMarkers, toggleDrawing }) => {
  const title = drawingMode !== drawToolConfig.DRAWING_MODE.NONE ? 'Eindig' :
    (shapeMarkers > 0 ? 'Opnieuw' : 'Begin');
  const label = title === 'Begin' ? '' : title;

  return (<button
    className={`
      toggle-drawing
      ${drawingMode !== drawToolConfig.DRAWING_MODE.NONE || shapeMarkers > 0 ? 'toggle-drawing--expanded' : 'toggle-drawing--collapsed'}
    `}
    onClick={() => toggleDrawing(shapeMarkers)}
    title={`${title} meten en intekenen`}
  >
    <IconMeasure className="toggle-drawing__icon" />
    <span className="toggle-drawing__label">
      {label}
    </span>
  </button>
  );
};

ToggleDrawing.propTypes = {
  toggleDrawing: PropTypes.func.isRequired,
  drawingMode: PropTypes.string.isRequired,
  shapeMarkers: PropTypes.number.isRequired
};

export default ToggleDrawing;
