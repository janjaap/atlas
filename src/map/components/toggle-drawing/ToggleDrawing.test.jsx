import React from 'react';
import { shallow } from 'enzyme';

import ToggleDrawing from './ToggleDrawing';
import drawToolConfig from '../../services/draw-tool/draw-tool.config';

describe('ToggleDrawing', () => {
  let toggleDrawing;

  beforeEach(() => {
    toggleDrawing = jest.fn();
  });

  it('should trigger toggle drawing on when clicked', () => {
    const wrapper = shallow(
      <ToggleDrawing
        drawingMode={drawToolConfig.DRAWING_MODE.NONE}
        shapeMarkers={0}
        toggleDrawing={toggleDrawing}
      />
    );
    wrapper.find('button').at(0).simulate('click');
    expect(toggleDrawing).toHaveBeenCalled();
  });

  it('should render with drawing mode none and no markers', () => {
    const wrapper = shallow(
      <ToggleDrawing
        drawingMode={drawToolConfig.DRAWING_MODE.NONE}
        shapeMarkers={0}
        toggleDrawing={toggleDrawing}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with drawing mode none and 3 markers showing "Opnieuw tekenen"', () => {
    const wrapper = shallow(
      <ToggleDrawing
        drawingMode={drawToolConfig.DRAWING_MODE.NONE}
        shapeMarkers={3}
        toggleDrawing={toggleDrawing}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with drawing mode draw and 3 markers showing "Eindig tekenen"', () => {
    const wrapper = shallow(
      <ToggleDrawing
        drawingMode={drawToolConfig.DRAWING_MODE.DRAW}
        shapeMarkers={3}
        toggleDrawing={toggleDrawing}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
