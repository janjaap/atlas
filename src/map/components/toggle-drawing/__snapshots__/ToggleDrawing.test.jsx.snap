// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleDrawing should render with drawing mode draw and 3 markers showing "Eindig tekenen" 1`] = `
<button
  className="
      toggle-drawing
      toggle-drawing--expanded
    "
  onClick={[Function]}
  title="Eindig meten en intekenen"
>
  <Component
    className="toggle-drawing__icon"
  />
  <span
    className="toggle-drawing__label"
  >
    Eindig
  </span>
</button>
`;

exports[`ToggleDrawing should render with drawing mode none and 3 markers showing "Opnieuw tekenen" 1`] = `
<button
  className="
      toggle-drawing
      toggle-drawing--expanded
    "
  onClick={[Function]}
  title="Opnieuw meten en intekenen"
>
  <Component
    className="toggle-drawing__icon"
  />
  <span
    className="toggle-drawing__label"
  >
    Opnieuw
  </span>
</button>
`;

exports[`ToggleDrawing should render with drawing mode none and no markers 1`] = `
<button
  className="
      toggle-drawing
      toggle-drawing--collapsed
    "
  onClick={[Function]}
  title="Begin meten en intekenen"
>
  <Component
    className="toggle-drawing__icon"
  />
  <span
    className="toggle-drawing__label"
  />
</button>
`;
